From 4644bd62b2dd48b3153879abce35fa3eb27fe3d3 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Mon, 4 Nov 2019 15:56:42 +0700
Subject: [PATCH 152/328] arm64: dts: renesas: hihope-common: Define and enable
 Bluetooth using PCA9654.

This patch defines and enables pin BT_REG_ON in PCA9654 (I/O Expander)
for Bluetooth device.

Signed-off-by: Khoi Nguyen <khoi.nguyen.aj@renesas.com>
Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/hihope-common.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/hihope-common.dtsi b/arch/arm64/boot/dts/renesas/hihope-common.dtsi
index 3fb066b..557ae00 100644
--- a/arch/arm64/boot/dts/renesas/hihope-common.dtsi
+++ b/arch/arm64/boot/dts/renesas/hihope-common.dtsi
@@ -400,6 +400,13 @@
 	status = "okay";
 	clock-frequency = <400000>;
 
+	pca9654: gpio@20 {
+		compatible = "onnn,pca9654";
+		reg = <0x20>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+
 	versaclock5: clock-generator@6a {
 		compatible = "idt,5p49v5923";
 		reg = <0x6a>;
@@ -488,3 +495,12 @@
 
 	status = "okay";
 };
+
+&pca9654 {
+	bluetooth-en-gpio {
+		gpio-hog;
+		gpios = <2 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "bluetooth-en-gpio";
+	};
+};
-- 
1.9.1

